packages/tax/src/services/tax-module-service.ts

```diff
@@ -166,4 +167,145 @@ export default class TaxModuleService<
   ) {
     return await this.taxRateRuleService_.create(data, sharedContext)
   }
+
+  async getTaxLines(
+    items: (TaxTypes.TaxableItemDTO | TaxTypes.TaxableShippingDTO)[],
+    calculationContext: TaxTypes.TaxCalculationContext,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<(TaxTypes.ItemTaxLineDTO | TaxTypes.ShippingTaxLineDTO)[]> {
+    const regions = await this.taxRegionService_.list(
+      {
+        $or: [
```

Comment:
Wouldn't 
```
{
  country_code: calculationContext.address.country_code
}
```

have the same result? Or is the country optional, so we would like to only fetch by province if it's not provided?

Reply:
The idea is that we as a maximum only get the country region and the province region. That maps to:
1. A region where the country code is set and the province code is not set.
2. A region where both the country code and the province code is set.

If we just listed all regions with the country code we would get a all province regions too. I.e. NY in US should only give:
- US
- US, NY

Your suggestion would give:
- US
- US, NY
- US, CA
- etc. etc.

Reply:
Yes you are right, understood what the intention is 👍